src/transformState.js

```diff
@@ -5,7 +5,24 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+// write code here
+  actions.forEach(stage => {
+    if (stage.type === 'addProperties') {
+      for (const keyToAdd in stage.extraData) {
+        state[keyToAdd] = stage.extraData[keyToAdd];
+      }
+    } else if (stage.type === 'removeProperties') {
+      stage.keysToRemove.forEach(keyToRemove => {
```

Comment:
There is no actual need in `else` statement here. You will always have exactly one "type" and it will obly work with one if statement.

```suggestion
    }
    if (stage.type === 'removeProperties') {
      stage.keysToRemove.forEach(keyToRemove => {
```